python/grass/script/setup.py

```diff
@@ -24,17 +24,24 @@
     # query GRASS itself for its GISBASE
     # (with fixes for specific platforms)
     # needs to be edited by the user
-    grass8bin = 'grass'
-    if sys.platform.startswith('win'):
+    grass8bin = "grass"
+    if sys.platform.startswith("win"):
         # MS Windows
-        grass8bin = r'C:\OSGeo4W\bin\grass.bat'
+        grass8bin = r"C:\OSGeo4W\bin\grass.bat"
         # uncomment when using standalone WinGRASS installer
         # grass8bin = r'C:\Program Files (x86)\GRASS GIS 8.0.0\grass.bat'
         # this can be avoided if GRASS executable is added to PATH
-    elif sys.platform == 'darwin':
+    elif sys.platform == "darwin":
         # Mac OS X
-        # TODO: this have to be checked, maybe unix way is good enough
-        grass8bin = '/Applications/GRASS/GRASS-8.0.app/'
+        mac_gisbase = "/Applications/GRASS-8.0.app/Contents/Resources"
+        usr_module_dir = "~/Library/GRASS/8.0/Modules"
+        grass8bin = f"{mac_gisbase}/bin/grass"
+        os.environ["PATH"] = (
+            f"{mac_gisbase}/bin:{mac_gisbase}/scripts:"
+            f"{usr_module_dir}/bin:{usr_module_dir}/scripts:"
```

Comment:
Ideally, this should be resolved by the init function. The idea is that all what you need to call the init function is one call to the binary. (Ideally, you would not needed even that with grass package on (Python) path.)